### PR TITLE
load more data added & btn styled

### PR DESCRIPTION
**Pull request by pjmiles**

### CODE REVIEW

File: src/components/display/DisplayImage.js [hunk 1]
```diff
@@ -2,42 +2,37 @@ import axiosInstance from "../api/axios";
...
-        console.log(error);
-      }
-    };
     showImage();
   }, []);
 
+  const loadMore = () => {
+    showImage();
+  };
+
```
Comment (Asylcreek) [inline src/components/display/DisplayImage.js:26]: I believe your `showImage` in one `useEffect` with the right amount of dependencies can handle loading the initial images and loading more images.

Hints
1. `perPage` should be a state variable
2. The `useEffect` deps should contain the `perPage` state variable
3. Your `loadMore` function should update the `perPage` state variable

File: src/components/display/DisplayImage.js [hunk 2]
```diff
@@ -2,42 +2,37 @@ import axiosInstance from "../api/axios";
...
-  // const [showMore, setShowMore] = useState(5);
+
+  const showImage = async () => {
+    try {
+      const res = await axiosInstance.get(`?&offset=${perPage}`);
+      setImages(res.data.results);
+    } catch (error) {
+      console.log(error);
+    }
+    perPage += 5;
```
Comment (Asylcreek) [inline src/components/display/DisplayImage.js:16]: I think you should use the length of `res.data.results` to update this

**Review by Asylcreek — CHANGES_REQUESTED** [review]
Why is the `DisplayImage` component that is handling the fetching of images, when you are clearly passing `images` and `setImages` to it ... should it not be whatever component that is sending those props that should handing the data fetching?

File: src/components/display/DisplayImage.js (hunk 1 above)
Comment (pjmiles) [reply src/components/display/DisplayImage.js:26]: ok I will adjust this.

File: src/components/display/DisplayImage.js (hunk 2 above)
Comment (pjmiles) [reply src/components/display/DisplayImage.js:16]: I dont get this length part? am I adding length method to get the response from the API? also what's the effect of this?

Comment (Asylcreek) [reply src/components/display/DisplayImage.js:16]: Imagine if there are only two pages, and there are 5 results per page.

If at any time there are only 6 results in the dB, that means your backend would only return 1 result for the second page, using hard coded `perPage += 5` doesn't account for a situation, using the length of the array you're getting from the backend does.

Comment (pjmiles) [reply src/components/display/DisplayImage.js:16]: oh ok. I will adjust this.

File: src/components/display/DisplayImage.js [hunk 3]
```diff
@@ -2,60 +2,63 @@ import axiosInstance from "../api/axios";
...
-        setImages(res.data.results);
-        console.log(res);
-      } catch (error) {
-        console.log(error);
+        const { data } = await axiosInstance.get(`?&offset=${perPage}`);
+        console.log(data)
+        setImages(data.results);
+        if(!data.next){
+          setErr("No more images")
+        }
```
Comment (Asylcreek) [inline src/components/display/DisplayImage.js:16]: Should the user see this as an error?

File: src/components/display/DisplayImage.js [hunk 4]
```diff
@@ -2,60 +2,63 @@ import axiosInstance from "../api/axios";
...
+      } catch {
+        setErr("Error showing images");
       }
     };
     showImage();
-  }, []);
+  }, [setImages, perPage, setErr]);
+
+  const loadMore = () => {
+    setPerPage((more) => more + 1); //not perfect yet
```
Comment (Asylcreek) [inline src/components/display/DisplayImage.js:25]: `more` is not a good name ... I went looking for where `more` was created.

And `perPage` is a string, and you are trying to add a number to it

File: src/components/display/DisplayImage.js [hunk 5]
```diff
@@ -2,60 +2,63 @@ import axiosInstance from "../api/axios";
...
-    // }
-  };
-
   return (
     <>
       <div className="display-section">
         <div className="image-section">
+          <div className="error-container">
+            <div className="error-display">{err}</div>
+          </div>
```
Comment (Asylcreek) [inline src/components/display/DisplayImage.js:43]: Should this not be conditionally rendered?

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: src/components/header/Header.js [hunk 6]
```diff
@@ -12,22 +12,24 @@ const Header = ({ setImages, setErr }) => {
...
+      } else {
+        setErr("");
       }
       setImages(data.results);
     } catch (error) {
       console.log(error);
     }
   };
 
+  const handleSearch = (e) => {
```
Comment (Asylcreek) [inline src/components/header/Header.js:29]:
```suggestion
  const handleSearchChange = (e) => {
```
or
```suggestion
  const handleChange = (e) => {
```

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: src/components/display/DisplayImage.js (hunk 3 above)
Comment (pjmiles) [reply src/components/display/DisplayImage.js:18]: oh will be adjusted

File: src/components/display/DisplayImage.js (hunk 5 above)
Comment (pjmiles) [reply src/components/display/DisplayImage.js]: adjusted

File: src/components/display/DisplayImage.js (hunk 4 above)
Comment (pjmiles) [reply src/components/display/DisplayImage.js:29]: I will change this to page

Comment (pjmiles) [reply src/components/display/DisplayImage.js:29]: I am still working on getting the paginated part rightly from the api cause the 1 added loading more for me presently but not checking through the entire images as I preferred

**Review by Asylcreek — APPROVED** [review]

**Pull request merged**